File: app/src/main/java/net/oliversne/sportsapp/MainActivity.java
package net.oliversne.sportsapp;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;

import net.oliversne.sportsapp.Adapters.SportAdapter;
import net.oliversne.sportsapp.Models.Sport;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {
    //Adapter View
    private RecyclerView recyclerView;
    //Data Source
    private List<Sport> sportList;
    //Adapter
    private SportAdapter adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        recyclerView = findViewById(R.id.recycler_view);
        sportList = new ArrayList<>();

        Sport s1 = new Sport("Football", R.drawable.football);
        Sport s2 = new Sport("Basketball", R.drawable.basketball);
        Sport s3 = new Sport("Tennis", R.drawable.tennis);
        Sport s4 = new Sport("Volleyball", R.drawable.volley);
        Sport s5 = new Sport("Ping Pong", R.drawable.ping);

        sportList.add(s1);
        sportList.add(s2);
        sportList.add(s3);
        sportList.add(s4);
        sportList.add(s5);

        adapter = new SportAdapter(sportList);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        recyclerView.setAdapter(adapter);
    }
}

File: app/src/main/java/com/adrianhelo/sportsapp/ui/MainActivity.kt
package com.adrianhelo.sportsapp.ui

import android.icu.lang.UCharacter.VerticalOrientation
import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.adrianhelo.sportsapp.R
import com.adrianhelo.sportsapp.adapter.SportsAdapter
import com.adrianhelo.sportsapp.model.SportsModel

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)

        // 1.- ListView
        val listView: RecyclerView = findViewById(R.id.activity_main_recycler_view)
        listView.layoutManager = LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false)

        // 2.- Data Source
        val sportList = ArrayList<SportsModel>()
        val sport1 = SportsModel("Football", R.drawable.football)
        val sport2 = SportsModel("Tennis", R.drawable.tennis)
        val sport3 = SportsModel("Basketball", R.drawable.basketball)
        val sport4 = SportsModel("Ping Pong", R.drawable.ping)
        val sport5 = SportsModel("Volley", R.drawable.volley)
        sportList.add(sport1)
        sportList.add(sport2)
        sportList.add(sport3)
        sportList.add(sport4)
        sportList.add(sport5)

        // 3.- Adapter
        val sportsAdapter = SportsAdapter(sportList)
        listView.adapter = sportsAdapter
    }
}